### PR TITLE
Add Seeed XIAO RP2350

### PR DESCRIPTION
**Pull request by tannewt**

### CODE REVIEW

File: ports/raspberrypi/boards/seeeduino_xiao_rp2350/pins.c [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+    CIRCUITPYTHON_BOARD_DICT_STANDARD_ITEMS
+
+    { MP_ROM_QSTR(MP_QSTR_A0), MP_ROM_PTR(&pin_GPIO26) },
+    { MP_ROM_QSTR(MP_QSTR_D0), MP_ROM_PTR(&pin_GPIO26) },
+
+    { MP_ROM_QSTR(MP_QSTR_A1), MP_ROM_PTR(&pin_GPIO27) },
+    { MP_ROM_QSTR(MP_QSTR_D1), MP_ROM_PTR(&pin_GPIO27) },
+
+    { MP_ROM_QSTR(MP_QSTR_A2), MP_ROM_PTR(&pin_GPIO28) },
+    { MP_ROM_QSTR(MP_QSTR_D2), MP_ROM_PTR(&pin_GPIO28) },
```
Comment (deshipu) [inline ports/raspberrypi/boards/seeeduino_xiao_rp2350/pins.c:19]: D3 pin is missing

Comment (deshipu) [reply ports/raspberrypi/boards/seeeduino_xiao_rp2350/pins.c:19]:
```c
    { MP_ROM_QSTR(MP_QSTR_A3), MP_ROM_PTR(&pin_GPIO5) },
    { MP_ROM_QSTR(MP_QSTR_D3), MP_ROM_PTR(&pin_GPIO5) },
```

File: ports/raspberrypi/boards/seeeduino_xiao_rp2350/pins.c [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+
+    { MP_ROM_QSTR(MP_QSTR_BAT_ADC_EN), MP_ROM_PTR(&pin_GPIO19) },
+
+    { MP_ROM_QSTR(MP_QSTR_A3), MP_ROM_PTR(&pin_GPIO29) },
+    { MP_ROM_QSTR(MP_QSTR_VOLTAGE_MONITOR), MP_ROM_PTR(&pin_GPIO29) },
```
Comment (deshipu) [inline ports/raspberrypi/boards/seeeduino_xiao_rp2350/pins.c:24]: I think we also want GPIO19 here for enabling the battery monitoring, it's BAT_ADC_EN on the schematic.

File: ports/raspberrypi/boards/seeeduino_xiao_rp2350/pins.c (hunk 1 above)
Comment (tannewt) [reply ports/raspberrypi/boards/seeeduino_xiao_rp2350/pins.c:19]: Added!

**Comment by tannewt** [comment]
I got a PID from the Seeed folks and updated here. The docs build failure is separate.

**Review by dhalbert — APPROVED** [review]
:+1:

**Pull request merged**